microsoft/testsuites/ltp/ltp.py

```diff
@@ -0,0 +1,357 @@
+    BUILD_REQUIRED_DISK_SIZE_IN_GB = 2
+    LTP_RESULT_PATH = "/opt/ltp/ltp-results.log"
+    LTP_OUTPUT_PATH = "/opt/ltp/ltp-output.log"
+    LTP_SKIP_FILE = "/opt/ltp/skipfile"
+
+    @property
+    def command(self) -> str:
+        return "/opt/ltp/runltp"
+
+    @property
+    def dependencies(self) -> List[Type[Tool]]:
+        return [Make, Gcc, Git]
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def run_test(
+        self,
+        environment: Environment,
+        ltp_tests: List[str],
+        skip_tests: List[str],
+        log_path: str,
+        drive_name: Optional[str] = None,
+    ) -> List[LtpResult]:
+        # tests cannot be empty
+        assert_that(ltp_tests, "ltp_tests cannot be empty").is_not_empty()
+        ls = self.node.tools[Ls]
+        rm = self.node.tools[Rm]
+
+        # remove skipfile if it exists
+        if ls.path_exists(self.LTP_SKIP_FILE):
+            self._log.debug(f"Removing skipfile: {self.LTP_SKIP_FILE}")
+            rm.remove_file(self.LTP_SKIP_FILE, sudo=True)
+
+        # remove results file if it exists
+        if ls.path_exists(self.LTP_RESULT_PATH, sudo=True):
+            self._log.debug(f"Removing {self.LTP_RESULT_PATH}")
+            rm.remove_file(self.LTP_RESULT_PATH, sudo=True)
+
+        # remove output file if it exists
+        if ls.path_exists(self.LTP_OUTPUT_PATH, sudo=True):
+            self._log.debug(f"Removing {self.LTP_OUTPUT_PATH}")
+            rm.remove_file(self.LTP_OUTPUT_PATH, sudo=True)
+
+        # add parameters for the test logging
+        parameters = f"-p -q -l {self.LTP_RESULT_PATH} -o {self.LTP_OUTPUT_PATH} "
+
+        # add the list of tests to run
+        parameters += f"-f {','.join(ltp_tests)} "
+
+        # add logging and output file parameter
+        if drive_name:
+            parameters += f"-z {drive_name} "
+
+        # add the list of skip tests to run
+        if len(skip_tests) > 0:
+            # write skip test to skipfile with newline separator
+            skip_file_value = "\n".join(skip_tests)
+            self.node.tools[Echo].write_to_file(
+                skip_file_value, PurePosixPath(self.LTP_SKIP_FILE), sudo=True
+            )
+            parameters += f"-S {self.LTP_SKIP_FILE} "
+
+        # Minimum 4M swap space is needed by some mmp test
+        if self.node.tools[Free].get_swap_size() < 4:
+            self.node.tools[Swap].create_swap()
+
+        # run ltp tests
+        self.run(parameters, sudo=True, force_run=True, timeout=12000)
+
+        # write output to log path
+        self.node.shell.copy_back(
+            PurePosixPath(self.LTP_OUTPUT_PATH), PurePath(log_path) / "ltp-output.log"
+        )
+
+        # write results to log path
+        self.node.shell.copy_back(
+            PurePosixPath(self.LTP_RESULT_PATH), PurePath(log_path) / "ltp-results.log"
+        )
+
+        # parse results
+        results = self._parse_results(self.LTP_RESULT_PATH)
```

Comment:
The file is copied back in previous lines, it doesn't need to read from remote again.